### ISSUE TITLE
Chunks with multiple passes are not supported.

### ISSUE THREAD

**Issue by azasisgod** [MiloszKrajewski/lz4net#6]
Hello..

I'm sending byte[] through internet and constructing it to the other side. When I tried to decompress I get the error "Chunks with multiple passes are not supported" . The lz4stream is able to decompress the first 65535 bytes then this message is shown. Any idea/hints where the problem would be? It works for small files no problem. Using highcompress and blockSize default value.

Some code below:
Client Side:
int totalcompressbytes = 0;
int CompressReadByteInt = 0;
byte[] CompressReadBuffer = new byte[1024 \* 16];
MemoryStream ms = new MemoryStream();
using (LZ4.LZ4Stream fslz4 = new LZ4.LZ4Stream(ms, System.IO.Compression.CompressionMode.Compress, true))
                {
                    while ((CompressReadByteInt = fs.Read(CompressReadBuffer, 0, CompressReadBuffer.Length)) > 0)
                    {
                        fslz4.Write(CompressReadBuffer, 0, CompressReadByteInt);
                        fslz4.Flush();
                        ms.Position = 0;
                        byte[] b = ms.ToArray();
                        totalcompressbytes += b.Length;
                        bool Complete = false;
                        if (CompressReadBuffer.Length > CompressReadByteInt)
                        {
                            Complete = true;
                        }
                        int returnvalue = (int)Serv.SendFile(values[0], values[1], System.Environment.MachineName, System.Environment.UserName, System.Security.Principal.WindowsIdentity.GetCurrent().User.AccountDomainSid.Value, values[2], fi.Name, fi.Length, b, 0, b.Length, totalcompressbytes - b.Length , BackupName, Directory, fi.LastWriteTimeUtc,Complete);
                }
}

Some Code Server Side:
public SendFileEnum SendFile(string Username, string Password, string ComputerName, string WindowsUser, string SID, string License, string filename, long filelength, byte[] data, int offset, int count, long position, string BackupName, string _Directory, DateTime LastWriteUTC, bool Complete)
{
try { fs = fi.OpenWrite(); }
catch { fs.Close(); return SendFileEnum.FALSE; }
fs.Position = position;
try { fs.Write(data, offset, count); }
catch (Exception ex) { fs.Close(); System.IO.File.AppendAllText(@"err.txt", ex.ToString()); return SendFileEnum.FALSE; }

}

**Comment by azasisgod** [comment]
I would guess that this might be because i'm calling flush, which the bytes read/flushed are less than the blocksize but in this case is file is also less then 1mb. 

Thanks.

**Comment by MiloszKrajewski** [comment]
I can't read your code. It's terribly formatted so it's hard to spot problem, but:
Number of passes is encoded in chuck header. It is always 1, LZ4 never writes any other value than 1. Therefore if reader get something different it means you do something wrong. your stream is corrupted or you just read the wrong stream. For example, I cannot see LZ4Stream anywhere is server code.
There is a unit test with server & client so use can use it as a starting point.

**Comment by azasisgod** [comment]
Hello Milosz,

I have fixed the issue. I was flushing the lz4stream, every 16*1024 bytes, and apparently that was before the block size was hit. I don't know if this was the problem or not, correct me if it isn't. Regardless, the current approach seems much better to me, as I get less overhead and better compression. I'm guessing, based on my tests, that during the flush is when the lz4stream really compress. 

Just so if anyone has the same issue, I have changed my code to monitor when lz4stream does flushes data to memorystream and then read the memorystream by 16k bytes and time. Flushes happens when bytesread go over the blocksize or at the end of reading the file if it's less than 1mb. If you aren't using "using blocks" then it's good to flush/disposing to get last data.

In addition, i'm not doing any decompressing at the server side. Saving the file as it is, already compressed. 

Thank for the for great work.
